### ISSUE TITLE
counts_rpkm in chapter 1 doesn't work with 64-bit NumPy builds, too

### ISSUE THREAD

**Issue by capissimo** [elegant-scipy/elegant-scipy#325]
Pictures 1-13 and 1-15 cannot be reproduced due to this:

c:\python36\lib\site-packages\ipykernel_launcher.py:1: RuntimeWarning: invalid value encountered in log
  """Entry point for launching an IPython kernel.
c:\python36\lib\site-packages\numpy\lib\function_base.py:4274: RuntimeWarning: Invalid value encountered in percentile
  interpolation=interpolation)
c:\python36\lib\site-packages\matplotlib\cbook\__init__.py:1856: RuntimeWarning: invalid value encountered in less_equal
  wiskhi = np.compress(x <= hival, x)
c:\python36\lib\site-packages\matplotlib\cbook\__init__.py:1863: RuntimeWarning: invalid value encountered in greater_equal
  wisklo = np.compress(x >= loval, x)
c:\python36\lib\site-packages\matplotlib\cbook\__init__.py:1871: RuntimeWarning: invalid value encountered in less
  np.compress(x < stats['whislo'], x),
c:\python36\lib\site-packages\matplotlib\cbook\__init__.py:1872: RuntimeWarning: invalid value encountered in greater
  np.compress(x > stats['whishi'], x)

I have Win10 64/Python 3.6.4/NumPy 1.14.2 mkl

**Comment by jni** [comment]
@capissimo interesting! Thanks for the report, we'll check it out and report back. Maybe NumPy 1.14 changed something...

**Comment by capissimo** [comment]
With the previous version (1.13.1) the outcome was the same. I'm running the examples in a Jupyter notebook. Maybe that's the reason...

**Comment by jni** [comment]
No, we build the whole book with the notebook, so I doubt that's the reason. Remind me, do we set the random seed and did you set yours?

**Comment by capissimo** [comment]
Yes, np.random.seed(seed=7). It was in the subsection 'Normalizing library size between samples', just before building the plot in Figure 1-7.

**Comment by jni** [comment]
Hmmm. So the figure works for me *but* the genes are not the same as in the book! Grrr. Looks like we need to figure out what's changing between runs or between systems. Working on it! Thank you again!

**Comment by jni** [comment]
Hmmm, `pd.Index.intersection`, I wonder if that's changed between pandas versions. Highly suspect.

**Comment by capissimo** [comment]
Just to be sure, updated pandas. No go.

**Comment by jni** [comment]
Gah: https://pandas.pydata.org/pandas-docs/version/0.20/whatsnew.html#index-intersection-and-inner-join-now-preserve-the-order-of-the-left-index

**Comment by jni** [comment]
@capissimo that explains my error at least. We'll have to update the book for it. Yours seems different. Would you like to chat at https://gitter.im/elegant-scipy/bugs? I have 15 mins now, or we could make some other time.

**Comment by capissimo** [comment]
But how does this relate to log_counts later for fig 1-13? Oddly enough, after updating pandas plots started to look differently(

**Comment by capissimo** [comment]
Checked count_rpkm fun for np.sum(np.isnan(array)) not being less 0

print(np.sum(np.isnan(L))) eq 0 
print(np.sum(np.isnan(C))) eq 0 
print(np.sum(np.isnan(normed))) eq 0

**Comment by jni** [comment]
As mentioned on gitter: what about np.sum(count_rpkm < 0)? Or < (-1)?

**Comment by capissimo** [comment]
2018-03-28 12:55 GMT+10:00 Juan Nunez-Iglesias <notifications@github.com>:

**Comment by capissimo** [comment]
I collected the 1st chapter's material in a Jupyter notebook for you to see
what's going on on my side.


2018-03-28 13:00 GMT+10:00 Андрей Логунов <laborhm@gmail.com>:

**Comment by jni** [comment]
@capissimo that's where your problem is. I think you must have made a mistake in creating `counts_rpkm` somewhere... Can you put your notebook on gist.github.com?

<img width="388" alt="screen shot 2018-03-28 at 10 24 48 am" src="https://user-images.githubusercontent.com/492549/38035468-7dadb914-3272-11e8-9759-e7acfa3858b2.png">

**Comment by capissimo** [comment]
Yes, I did as you told.

2018-03-29 0:27 GMT+10:00 Juan Nunez-Iglesias <notifications@github.com>:

**Comment by capissimo** [comment]
Ple notice, how the plots look different after I upgraded pandas((

2018-03-29 0:37 GMT+10:00 Андрей Логунов <laborhm@gmail.com>:

**Comment by jni** [comment]
@capissimo I don't know where your notebook is? I don't see a link

**Comment by capissimo** [comment]
Pardon, here it is
https://gist.github.com/capissimo/16306b04fd4039d48ba6df67857fbf88

2018-03-29 0:48 GMT+10:00 Juan Nunez-Iglesias <notifications@github.com>:

**Comment by jni** [comment]
@capissimo you must have run and re-run certain "play" cells and gotten the notebook into a bad state. I've run the notebook from scratch and things work! So try "Restart and run all" in the notebook and I hope that will fix your problem?

btw I can't tell you how happy it makes me to see it all in Russian. =D

**Comment by jni** [comment]
Also btw if there are any parts that you thought were unclear or "ugly", please let us know!

**Comment by capissimo** [comment]
Hi, Juan,

Found a couple of places, where to update.

p. 79, 153:   Newer versions of Networkx do not have edges_iter()
g.edges_iter() --> g.edges()   (OR: g.edges)


p. 152:  A typo. Using a tuple causes error, so change to list repr, () -->
[]
def threshold_graph(g, t):
to_remove = ((u, v) for (u, v, d) in g.edges(data=True)
             if d['weight'] > t)
g.remove_edges_from(to_remove)

def threshold_graph(g, t):
to_remove = [(u, v) for (u, v, d) in g.edges(data=True)
                     if d['weight'] > t]
g.remove_edges_from(to_remove)

2018-03-29 1:13 GMT+10:00 Juan Nunez-Iglesias <notifications@github.com>:

**Comment by capissimo** [comment]
I still can't get how I should use intersection to get things straight.

2018-03-29 9:22 GMT+10:00 Андрей Логунов <laborhm@gmail.com>:

**Comment by capissimo** [comment]
In general I found the book very interesting. What it makes more valuable
is the absence of practical stuff on SciPy out there.
Some may argue that a vague genetic analysis procedure and FFT concepts
sort of downgrades it.
My point is that this is science.
Some will be motivated to go and study up on RNA-seq, FFT, etc.,
but the majority, I'm sure, will take home the underlying priciple and use
it in their own field.
Personally, I like most the last part on combination of scientific analysis
with functional programming.
And one last thing.
In all there two main ways how to write science-oriented computer books.
You either take a scientific problem and show how to solve it, or you take
a library function and show how to use it in practice.
SciPy now has a remarkable book written according to theformer, but still
lacks the latter.
Hope, we shall see a continuation))


2018-03-29 9:24 GMT+10:00 Андрей Логунов <laborhm@gmail.com>:

**Comment by capissimo** [comment]
SOLVED!
This has done the trick: N = np.sum(counts, axis=0).astype(float) in rpkm

**Comment by jni** [comment]
👏👏👏👏 🎉🎉🎉🎉

By the way, could you check whether the version in the PR works for you also?

```python
C = counts.astype(float)
N = np.sum(C, axis=0)
```
?

**Comment by capissimo** [comment]
Yes, it does work all right.


2018-04-01 11:45 GMT+10:00 Juan Nunez-Iglesias <notifications@github.com>:
